torch/csrc/jit/codegen/cuda/kernel_ir.cpp

```diff
@@ -328,16 +336,35 @@ void Scope::insert_after(Expr* ref, Expr* expr) {
       " after the reference: ",
       ref,
       " however the reference was not found in this scope.");
-  exprs_.insert(it + 1, expr);
+  insert(it + 1, expr);
+}
+
+void Scope::insert(size_t pos, Expr* expr) {
+  const auto it = exprs_.begin() + pos;
+  insert(it, expr);
+}
+
+void Scope::erase(std::vector<Expr*>::const_iterator pos) {
+  // Remove the scope of the expr if this is the scope
+  auto expr = *pos;
+  if (expr->scope() == this) {
+    expr->removeScope();
+  }
+  exprs_.erase(pos);
 }
 
 void Scope::erase(Expr* ref) {
```

Comment:
Do we really want/need such O(N) helper?

Reply:
Also, looks like good candidate for erase/remove idiom

Reply:
I didn't change the basic design of the API here, so, yes, it should be removed if not necessary, but that would be a different PR.